src/main/java/spireMapOverhaul/zones/mirror/MirrorZone.java

```diff
@@ -0,0 +1,148 @@
+        if (!previousMoveData.isEmpty()) {
+            AbstractDungeon.player.gameHandSize++;
+        }
+        previousMoveData.clear();
+        chosenMoveData = null;
+    }
+
+    public static void addMoveData(MirrorMoveData data) {
+        if (previousMoveData.isEmpty()) {
+            AbstractDungeon.player.gameHandSize--;
+        }
+        previousMoveData.add(data);
+    }
+
+    public static void putPixmap(String id, Pixmap pixmap) {
+        Pixmap pixmapHalf = new Pixmap(250, 190, pixmap.getFormat());
+        pixmapHalf.drawPixmap(pixmap,
+                0, 0, pixmap.getWidth(), pixmap.getHeight(),
+                0, 0, pixmapHalf.getWidth(), pixmapHalf.getHeight()
+        );
+        Texture texture = new Texture(pixmap);
+        Texture textureHalf = new Texture(pixmapHalf);
+        pixmapHalf.dispose();
+        pixmap.dispose();
+
+        largeTextures.put(id, texture);
+        smallTextures.put(id, textureHalf);
+        CustomCard.imgMap.put(MirrorMove.getPortraitId(id), textureHalf);
+    }
+
+    public static boolean textureMissing(String id) {
+        return !largeTextures.containsKey(id);
+    }
+
+    @Override
+    public AbstractZone copy() {
+        return new MirrorZone();
+    }
+
+    @Override
+    public Color getColor() {
+        return new Color(0.4f, 0.5f, 0.8f, 0.75f);
+    }
+
+    @Override
+    public void loadStrings() {
+        super.loadStrings();
+        if (SpireAnniversary6Mod.initializedStrings) {
+            MirrorMove.initializeDescriptionParts();
+        }
+    }
+
+    @Override
+    public void atBattleStartPreDraw() {
+        resetPixmap();
+        // Not calling resetMoveData() here because gameHandSize shouldn't increase when loading a save file
+        previousMoveData.clear();
+        chosenMoveData = null;
+
+        AbstractDungeon.actionManager.addToBottom(new ApplyPowerAction(AbstractDungeon.player, AbstractDungeon.player, new MirrorZonePower(AbstractDungeon.player)));
+    }
+
+    @Override
+    public void atTurnStart() {
+        if (!previousMoveData.isEmpty()) {
+            chosenMoveData = previousMoveData.get(AbstractDungeon.cardRandomRng.random(previousMoveData.size() - 1));
+            MirrorMove c = new MirrorMove();
+            c.applyPowers();
+            AbstractDungeon.actionManager.addToBottom(new MakeTempCardInHandAction(c));
+            AbstractPower pow = AbstractDungeon.player.getPower(MirrorZonePower.POWER_ID);
+            if (pow != null) {
+                pow.flash();
+            }
+        }
+    }
+
+    @Override
+    public void atTurnEnd() {
+        resetMoveData();
+    }
+
+    @Override
+    public void onVictory() {
+        resetPixmap();
+        resetMoveData();
+    }
+
+    @Override
+    public String getCombatText() {
+        return MirrorZonePower.DESCRIPTIONS[0];
+    }
+
+    @Override
+    public AbstractEvent forceEvent() {
+        return ModifiedEventRateZone.returnIfUnseen(WeirdMirrorsEvent.ID);
```

Comment:
This has changed, please merge the latest changes and change the return of forceEvent to string

Reply:
Fixed with https://github.com/erasels/SpireMapOverhaul/pull/82/commits/3f6b95c8bf744d7a55a1dc99f3deac7ff46609ce